graylog2-web-interface/src/components/users/UserForm.jsx

```diff
@@ -111,6 +111,9 @@ const UserForm = React.createClass({
       if (this.isPermitted(this.state.currentUser.permissions, ['users:list'])) {
         this.props.history.replaceState(null, Routes.SYSTEM.AUTHENTICATION.USERS.LIST);
       }
+      if (this.props.user.username === this.state.currentUser.username) {
+        CurrentUserStore.reload();
+      }
```

Comment:
I was wondering, shouldn't we do the same when updating permissions or roles? If so, maybe we should trigger the reload from the (ancient) `UsersStore`, or even take the time to convert it to a Reflux store. We should probably wait for that until 2.1 is out, though.